### ISSUE TITLE
ExtractArchive Crashes on non existing files

### ISSUE THREAD

**Issue by protree** [pyload/pyload#3277]
ExtractArchive Crashes on non existing files.

**Steps to reproduce**: Download .rar file, let it extract. Delete the rar file while keeping its entry in queque. restart pyload. ExtractArchive will crash with following log messages and won't extract any archives until entry is removed from queque and pyload is restarted:

> pyload_1          | 24.11.2018 17:56:43 INFO      ADDON ExtractArchive: Check package: packageRenamed
> pyload_1          | Exception in thread Thread-8:
> pyload_1          | Traceback (most recent call last):
> pyload_1          |   File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
> pyload_1          |     self.run()
> pyload_1          |   File "/opt/pyload/module/PluginThread.py", line 459, in run
> pyload_1          |     self.f(*self.args, **self.kwargs)
> pyload_1          |   File "/opt/pyload/module/plugins/hooks/ExtractArchive.py", line 178, in extract_queued
> pyload_1          |     if self.extract(packages, thread):  # @NOTE: check only if all gone fine, no failed reporting for now
> pyload_1          |   File "/opt/pyload/module/plugins/hooks/ExtractArchive.py", line 368, in extract
> pyload_1          |     if len(os.listdir(pack_dl_folder)) == 0:
> pyload_1          | OSError: [Errno 2] No such file or directory: '/opt/pyload/Downloads/Download/packageRenamed/'

**Comment by GammaC0de** [comment]
> Steps to reproduce: Download .rar file, let it extract. 
Delete the rar file while keeping its entry in queue. 

Have you deleted the .rar file before the extraction completed?

**Comment by protree** [comment]
No, I deleted the file after extraction finished. It also happens when you enable „delete packages after successful extraction“ in extractarchive settings.

**Comment by GammaC0de** [comment]
What is the package status after the process?

Debug log please..

**Comment by protree** [comment]
Following Situation with debug log:

2 packages:
- _oldPackageThatIsInQuequeButWasDeletedAtSomePoint_: ExtractArchive crashes on this package on restart, unfortunally I don't have logs how this package was downloaded and extracted
- _packageToExtract_: Package is added and downloaded but doesn't extract. I restarted pyload and ExtractArchive crashed again on _oldPackageThatIsInQuequeButWasDeletedAtSomePoint_. Next I deleted _oldPackageThatIsInQuequeButWasDeletedAtSomePoint_ from queque and restarted pyload. _packageToExtract_ then gets extracted after restart automatically. Extraction finishes without errors and .rar files get deleted automatically by ExtractArchive. After another restart of pyload, ExtractArchive now crashes on this package...

I didn't manage to reproduce this bug again starting from a clear queque, but it happens every now and then... If I am able to get another debug log I will send it to you...

For me it looks like ExtractArchive completely crashes if a file which it expects isn't available and doesn't recover from this situations until the "corrupted entry" in queque is deleted and pyload is restarted...

[pyload_debuglog.txt](https://github.com/pyload/pyload/files/2613160/pyload_debuglog.txt)

**Comment by GammaC0de** [comment]
This is not a debug log, you have to enable debug mode in pyload's configuration first

**Comment by GammaC0de** [comment]
Any update?

**Comment by protree** [comment]
Unfortunatly not... I tried to reproduce it again with debug log turned on, but without success. What I noticed was that in my setup (I run pyload inside a docker container on a server that shuts down every night) pyload is always killed by SIGKILL by docker-daemon on server shutdown or docker restart (after container update) because it doesn‘t respond to SIGTERM in 10 seconds in my setup, I don‘t know why yet. Maybe this leads to a currupted state of pyload ever now and then, but I have to look deeper into this.

This might take a while. Maybe I close this issue and reopen it when I have any updates?
